src/models/AbstractWidget.php

```diff
@@ -0,0 +1,126 @@
+<?php namespace Leean\Widgets\Models;
+
+use Leean\Acf\All;
+
+/**
+ * Class AbstractWidget.
+ *
+ * @package Leean\Widgets\Models
+ */
+class AbstractWidget extends \WP_Widget
+{
+	/**
+	 * The slug.
+	 *
+	 * @var bool|string
+	 */
+	private $_slug = '';
+
+	/**
+	 * AbstractWidget constructor.
+	 *
+	 * @param string $title       Widget title
+	 * @param string $description Widget description.
+	 * @param bool   $slug		  Widget slug (use class name if not supplied).
+	 */
+	public function __construct( $title, $description, $slug = false ) {
```

Comment:
ok no I mean my whole point is here since you are naming this class `Abstract` we are assuming this is an abstract. 

> That's the definition of an abstract class if you are not going to use it as such, you should removed the prefix abstract.

If you are using plain inheritance with no other property of an abstract class you should rename your class instead and remove the abstract keyword.

Also I'm not choosing what method should be abstract,  And yes and abstract class can va N methods with the implementation but requires at least 1 abstract method.